src/providers/ogr/qgsogrdataitems.cpp

```diff
@@ -118,28 +119,41 @@ QString QgsOgrLayerItem::layerName() const
 QList<QAction *> QgsOgrLayerItem::actions()
 {
   QList<QAction *> lst;
-  QAction *actionDeleteLayer = new QAction( tr( "Delete %1" ).arg( mName ), this );
+  // Messages are different for files and tables
+  QString message = mIsSubLayer ? QObject::tr( "Delete layer '%1'..." ).arg( mName ) : QObject::tr( "Delete file '%1'..." ).arg( mUri );
+  QAction *actionDeleteLayer = new QAction( message, this );
   connect( actionDeleteLayer, &QAction::triggered, this, &QgsOgrLayerItem::deleteLayer );
   lst.append( actionDeleteLayer );
   return lst;
 }
 
 void QgsOgrLayerItem::deleteLayer()
 {
-  if ( QMessageBox::question( nullptr, QObject::tr( "Delete Layer" ),
-                              QObject::tr( "Are you sure you want to delete %1?" ).arg( mName ),
+  // Messages are different for files and tables
+  QString title = mIsSubLayer ? QObject::tr( "Delete Layer" ) : QObject::tr( "Delete File" );
+  QString confirmMessage;
+  if ( mIsSubLayer )
+  {
+    confirmMessage = QObject::tr( "Are you sure you want to delete layer '%1'?" ).arg( mName );
```

Comment:
With the "delete layer" there is still a potential confusion with project layers, so I'd say "delete layer from datasource" to be clear.